src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -23,37 +23,39 @@ namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
 {
     public class SqlQueueClient : IQueueClient
     {
-        private readonly SqlConnectionWrapperFactory _sqlConnectionWrapperFactory;
         private readonly SchemaInformation _schemaInformation;
         private readonly ISqlRetryService _sqlRetryService;
         private readonly ILogger<SqlQueueClient> _logger;
 
         public SqlQueueClient(
-            SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
             SchemaInformation schemaInformation,
             ISqlRetryService sqlRetryService,
             ILogger<SqlQueueClient> logger)
         {
-            EnsureArg.IsNotNull(sqlConnectionWrapperFactory, nameof(sqlConnectionWrapperFactory));
-            EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
-            EnsureArg.IsNotNull(sqlRetryService, nameof(sqlRetryService));
-            EnsureArg.IsNotNull(logger, nameof(logger));
-
-            _sqlConnectionWrapperFactory = sqlConnectionWrapperFactory;
-            _schemaInformation = schemaInformation;
-            _sqlRetryService = sqlRetryService;
-            _logger = logger;
+            _schemaInformation = EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
+            _sqlRetryService = EnsureArg.IsNotNull(sqlRetryService, nameof(sqlRetryService));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        private SqlQueueClient(ISqlRetryService sqlRetryService, ILogger<SqlQueueClient> logger)
+        {
+            _sqlRetryService = EnsureArg.IsNotNull(sqlRetryService, nameof(sqlRetryService));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public static SqlQueueClient GetInstance(ISqlRetryService sqlRetryService)
+        {
+            return new SqlQueueClient(sqlRetryService, new NullLogger<SqlQueueClient>());
```

Comment:
NullLogger? I think this class was only supposed to be used in tests, isn't it?

Reply:
Removed